support-frontend/assets/pages/weekly-subscription-checkout/components/weeklyCheckoutForm.test.jsx

```diff
@@ -0,0 +1,108 @@
+
+const pageReducer = initialState => createWithDeliveryCheckoutReducer(
+  initialState.common.internationalisation.countryId,
+  GuardianWeekly,
+  'Annual',
+  formatMachineDate(new Date()),
+  initialState.page.checkout.productOption,
+  initialState.page.checkout.fulfilmentOption,
+);
+
+function setUpStore(initialState) {
+  return createStore(
+    combineReducers({ page: pageReducer(initialState), common: createCommonReducer(initialState.common) }),
+    initialState,
+    compose(applyMiddleware(thunk)),
+  );
+}
+
+function renderWithStore(
+  component,
+  {
+    initialState,
+    store = setUpStore(initialState),
+    ...renderOptions
+  } = {},
+) {
+  function Wrapper({ children }) {
+    return <Provider store={store}>{children}</Provider>;
+  }
+
+  return render(component, { wrapper: Wrapper, ...renderOptions });
+}
+
+describe('Guardian Weekly checkout form', () => {
+  // Suppress warnings related to our version of Redux and improper JSX
+  console.warn = jest.fn();
+  console.error = jest.fn();
+
+  let initialState;
+
+  beforeEach(() => {
+    initialState = {
+      page: {
+        checkout: {
+          product: 'Paper',
+          billingPeriod: 'Monthly',
+          productOption: 'NoProductOptions',
+          fulfilmentOption: 'Domestic',
+          productPrices: weeklyProducts,
+          formErrors: [],
+        },
+        billingAddress: {
+          fields: {
+            country: 'GB',
+            formErrors: [],
+          },
+        },
+        deliveryAddress: {
+          fields: {
+            country: 'GB',
+            formErrors: [],
+          },
+        },
+      },
+      common: {
+        internationalisation: {
+          countryGroupId: 'GBPCountries',
+          countryId: 'GB',
+          currencyId: 'GBP',
+        },
+      },
+    };
+
+    renderWithStore(<WeeklyCheckoutForm />, { initialState });
+  });
+
+  describe('Payment methods', () => {
+    it('shows the direct debit option when the currency is GBP and the delivery address is in the UK', async () => {
+      expect(await screen.queryByText('Direct debit')).toBeInTheDocument();
+    });
+
+    it('does not show the direct debit option when the delivery address is outside the UK', async () => {
+      const countrySelect = await screen.findByLabelText('Country');
+      fireEvent.change(countrySelect, { target: { value: 'DE' } });
+      expect(await screen.queryByText('Direct debit')).not.toBeInTheDocument();
+    });
+
+    it('does not show the direct debit option when the billing address is outside the UK', async () => {
+      const addressIsNotSame = await screen.findByRole('radio', { name: 'No' });
+      fireEvent.click(addressIsNotSame);
+      const allCountrySelects = await screen.findAllByLabelText('Country');
+      fireEvent.change(allCountrySelects[1], { target: { value: 'IE' } });
+      expect(await screen.queryByText('Direct debit')).not.toBeInTheDocument();
+    });
```

Comment:
Could test 'does not show the direct debit option when currency is EUR even if billing address and delivery address are in the UK'

Reply:
How would a user get into that scenario? Setting the delivery address automatically switches the payment currency, and if you enter the checkout from the EU landing page your delivery country is initially set to be Germany.